src/test/scala/za/co/absa/hyperdrive/trigger/api/rest/services/NotificationRuleValidationServiceTest.scala

```diff
@@ -18,7 +18,7 @@ package za.co.absa.hyperdrive.trigger.api.rest.services
 import org.mockito.ArgumentMatchers.{eq => eqTo, _}
 import org.mockito.Mockito._
 import org.scalatest.mockito.MockitoSugar
-import org.scalatest.{AsyncFlatSpec, BeforeAndAfter, Matchers}
+import org.scalatest.{AsyncFlatSpec, BeforeAndAfter, Matchers, ParallelTestExecution}
```

Comment:
I don't think this is the way to go. We have mocks that are used by all test cases, such as `workflowRepository`. If the test cases are allowed to run in parallel, that introduces race conditions.

I think we should just stick to the documentation and wrap the assertions into future transformers (e.g. `map`) instead of using `Await`. Then we can also use the default single-thread execution context and don't need to override it.

Example:
```
class FutureTest extends AsyncFlatSpec {
  "A Future" should "return the expected result" in {
    val f = Future {
      Thread.sleep(100)
      "Hello"
    }
    f.map { result =>
      assert(result == "Hello")
    }
  }
}
```

Reply:
I forgot to move it to every test.
But I did it as documentation says.